src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -793,9 +793,21 @@ export default class Fms {
      * @return {boolean}
      */
     replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
-
-        if (routeStringElements.length !== 3) {
+        var routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+
+        // if routeString contains an entry element, use that as the departure runway
+        // otherwise, keep the departure runway already defined
+        // complain if there is no departure runway defined
+        if (routeStringElements.length === 2 ) {
+            if(this.departureRunwayModel){
+                routeString = this.departureAirportModel.icao.toUpperCase() + this.departureRunwayModel.name + PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER + routeString;
```

Comment:
@erikquinn there has to be an easier way to do this stuff.  This inline string building is not maintainable and just looks gross.

Reply:
i agree, but idk what that would be... :/